lib/utils.js

```diff
@@ -19,6 +19,35 @@ function shallowCopy(src, dest) {
   return dest;
 }
 
+function take(source, fromRight) {
+  return fromRight ? source.pop() : source.shift();
+}
+
+/**
+ * Takes a list of arguments and maps them to a list of names
+ * in an alternating either a left-to-right, or right-to-left direction.
+ */
+function resolveArgs(argumentsObject, nameMap, rightToLeft) {
+  var result = {};
+
+  var takeFromRight = rightToLeft;
+  var args = Array.prototype.slice.call(argumentsObject);
+
+  while (nameMap.length) {
+    var value = null;
+    var name = take(nameMap, takeFromRight);
+
+    if (args.length > 0) {
+      value = take(args, takeFromRight);
+      takeFromRight = !takeFromRight;
```

Comment:
It's a hack. But I'm pretty certain that it should work since it covers the two most common mapping scenarios: `href, [options,] callback` and `[options,] callback`.

And as I wrote before, it's a hack and I have an idea of how to fix it long-term :) But I haven't tried using the `express-stormpath` module and linking this in, so will do that and see what it throws.